src/smart-components/user/add-user-to-group/add-user-to-group-wizard.js

```diff
@@ -0,0 +1,267 @@
+import GroupRowWrapper from '../../group/group-row-wrapper';
+import { sortable } from '@patternfly/react-table';
+import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
+import { shallowEqual, useSelector, useDispatch } from 'react-redux';
+import { createQueryParams } from '../../../helpers/shared/helpers';
+import messages from '../../../Messages';
+import pathnames from '../../../utilities/pathnames';
+import { applyPaginationToUrl, defaultAdminSettings, defaultSettings, isPaginationPresentInUrl } from '../../../helpers/shared/pagination';
+import { useIntl } from 'react-intl';
+import { TableToolbarView } from '../../../presentational-components/shared/table-toolbar-view';
+
+import { useFieldApi } from '@data-driven-forms/react-form-renderer';
+import PermissionsContext from '../../../utilities/permissions-context';
+import { addMembersToGroup, fetchGroups } from '../../../redux/actions/group-actions';
+import { useHistory } from 'react-router-dom';
+import { getUserGroups } from '../../../helpers/group/group-helper';
+
+const FORM_ID = 'add-user-to-group-form';
+
+const GroupTable = (conf) => {
+  const { input, user, setLength } = useFieldApi(conf);
+  const textFilterRef = useRef(null);
+  const { orgAdmin, userAccessAdministrator } = useContext(PermissionsContext);
+  const isAdmin = orgAdmin || userAccessAdministrator;
+  const dispatch = useDispatch();
+  const intl = useIntl();
+  const history = useHistory();
+
+  const columns = [
+    { title: intl.formatMessage(messages.name), key: 'name', transforms: [sortable] },
+    { title: intl.formatMessage(messages.roles) },
+    { title: intl.formatMessage(messages.members) },
+    { title: intl.formatMessage(messages.lastModified), key: 'modified', transforms: [sortable] },
+  ];
+
+  const { groups, meta, filters, isLoading } = useSelector(
+    ({ groupReducer: { groups, isLoading } }) => ({
+      groups: [...(groups?.data?.filter(({ platform_default, admin_default } = {}) => !(platform_default || admin_default)) || [])],
+      meta: groups?.pagination || groups?.meta,
+      filters: groups?.filters,
+      isLoading,
+    }),
+    shallowEqual
+  );
+  const [pagination, setPagination] = useState({ ...(orgAdmin ? defaultAdminSettings : defaultSettings), ...meta });
+  const [filterValue, setFilterValue] = useState(filters.name || '');
+  const [removeGroupsList, setRemoveGroupsList] = useState([]);
+  const [data, setData] = useState(undefined);
+
+  const setCheckedItems = (newSelection) => {
+    const rows = input.value || [];
+    const selection = newSelection(rows)
+      .filter(({ platform_default: isPlatformDefault, admin_default: isAdminDefault }) => !(isPlatformDefault || isAdminDefault))
+      .map(({ uuid, name }) => ({ uuid, label: name }));
+    setLength(selection.length);
+
+    input.onChange(selection);
+  };
+  const fetchData = (options) => dispatch(fetchGroups({ ...options, inModal: false }));
```

Comment:
there, we definitely want to set inModal to `true`, if set to `false`, the fetch rewrites pagination data for the Groups page, we don't want that. Maybe the name `inModal` is not descriptive enough. I'll open a ticket to rename it